destination_integration_test.go

```diff
@@ -0,0 +1,237 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cassandra
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/gocql/gocql"
+	"github.com/matryer/is"
+)
+
+const (
+	testKeyspace = "conduit_test"
+	testHost     = "127.0.0.1"
+	testPort     = "9042"
+)
+
+func TestDestination_Write(t *testing.T) {
+	is := is.New(t)
+	ctx := context.Background()
+
+	// simple connect to get a Cassandra session
+	session := simpleConnect(t, map[string]string{
+		"host": testHost,
+		"port": testPort,
+	})
+	// use the simple connect session to setup for the test
+	table := SetupTest(t, session)
+
+	destination := NewDestination()
+	err := destination.Configure(ctx, map[string]string{
+		"host":     testHost,
+		"port":     testPort,
+		"keyspace": testKeyspace,
+		"table":    table,
+	})
+	is.NoErr(err)
+	err = destination.Open(ctx)
+	is.NoErr(err)
+	defer func() {
+		err := destination.Teardown(ctx)
+		is.NoErr(err)
+	}()
+
+	testCases := []struct {
+		name   string
+		record sdk.Record
+	}{{
+		name: "snapshot operation to insert query",
+		record: sdk.Record{
+			Position:  sdk.Position("foo"),
+			Operation: sdk.OperationSnapshot,
+			Key:       sdk.StructuredData{"id1": "6", "id2": 6},
+			Payload: sdk.Change{
+				After: sdk.StructuredData{
+					"column1": 22,
+					"column2": false,
+					// match the precision that Cassandra uses for timestamp.
+					"column3": time.Now().UTC().Truncate(time.Millisecond),
+				},
+			},
+		},
+	}, {
+		name: "create operation to insert query",
+		record: sdk.Record{
+			Operation: sdk.OperationCreate,
+			Key:       sdk.StructuredData{"id1": "7", "id2": 7},
+			Payload: sdk.Change{
+				After: sdk.StructuredData{
+					"column1": 33,
+					"column2": true,
+					"column3": time.Now().UTC().Truncate(time.Millisecond),
+				},
+			},
+		},
+	}, {
+		name: "update operation",
+		record: sdk.Record{
+			Position:  sdk.Position("foo"),
+			Operation: sdk.OperationUpdate,
+			// this record is already in the table
+			Key: sdk.StructuredData{"id1": "1", "id2": 1},
+			Payload: sdk.Change{
+				After: sdk.StructuredData{
+					"column1": 44,
+					"column2": false,
+					"column3": time.Now().UTC().Truncate(time.Millisecond),
+				},
+			},
+		},
+	}, {
+		name: "delete operation",
+		record: sdk.Record{
+			Operation: sdk.OperationDelete,
+			// this record is already in the table
```

Comment:
I guess this is because we assume the previous tests passed?

Reply:
I updated the method doc to mention this